src/vm/gchelpers.cpp

```diff
@@ -1319,6 +1341,13 @@ void ErectWriteBarrierForMT(MethodTable **dst, MethodTable *ref)
     
     if (ref->Collectible())
     {
+#ifdef FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
+        if (SoftwareWriteWatch::IsEnabledForGCHeap())
+        {
```

Comment:
Sure, I'll ask around and try to get some context on what the scenarios are for this. It may be suboptimal if unnecessary, but it shouldn't be functionally incorrect to mark more pages as dirty, so it should be a safe change in that regard.